src/JestExt.ts

```diff
@@ -527,32 +569,54 @@ export class JestExt {
   public runTest = (fileName: string, identifier: string) => {
     const restart = this.jestProcessManager.numberOfProcesses > 0
     this.jestProcessManager.stopAll()
-    const program = this.resolvePathToJestBin()
-    if (!program) {
+    const launcher = this.resolveTestProgram()
+    if (!launcher) {
       console.log("Could not find Jest's CLI path")
       return
     }
 
     const escapedIdentifier = JSON.stringify(identifier).slice(1, -1)
 
-    const args = ['--runInBand', fileName, '--testNamePattern', escapedIdentifier]
+    let args = launcher.args
+    // These settings identify the test we would like to run.
+    args.push('--runInBand', fileName, '--testNamePattern', escapedIdentifier)
     if (this.pluginSettings.pathToConfig.length) {
       args.push('--config', this.pluginSettings.pathToConfig)
     }
 
     const port = Math.floor(Math.random() * 20000) + 10000
+    let runtimeArgs = ['--inspect-brk=' + port]
+    const env: any = {}
+
+    if (launcher.isCreateReactApp) {
+      // If the project has been created by create-react-app,
+      // things get a little bit complicated.
+      // Running `react-scripts test` starts Jest within a separately spawned
+      // process and we want to debug that process, not the outer
+      // `react-scripts` process.
+      // We therefore run `react-scripts --inspect-brk=... test`, because
+      // `react-scripts` interprets the arguments before `test` as launching
+      // arguments for generating the new node process.
+      args = runtimeArgs.concat(args)
+      runtimeArgs = []
+      // `react-scripts` automatically appends the `--watch` flag
+      // if CI isn't set.
+      env.CI = 'VSCode-test-debugger'
+    }
+
     const configuration = {
       name: 'TestRunner',
       type: 'node',
       request: 'launch',
-      program,
+      program: launcher.program,
       args,
-      runtimeArgs: ['--inspect-brk=' + port],
+      runtimeArgs: runtimeArgs,
       port,
       protocol: 'inspector',
       console: 'integratedTerminal',
       smartStep: true,
       sourceMaps: true,
+      env,
     }

```

Comment:
It'd be helpful to also let us override `runtimeExecutable`, for cases where we need to pass `--inspect-brk` through to a child process

(example use case: https://github.com/nevir/lib-boundless/blob/master/scripts/jest.js)

Reply:
That would be the plan, since that is also what we need for supporting create-react-app apps.

Reply:
Ah, sweet